src/renderer/fonts/caching_shaper.rs

```diff
@@ -32,6 +32,7 @@ pub struct CachingShaper {
     shape_context: ShapeContext,
     scale_factor: f32,
     fudge_factor: f32,
+    linespace: u64,
```

Comment:
According to `:h linespace`, it can also be negative, should likely be a `i64` instead:

> 	With some fonts there can be too much room between lines (to have space for ascents and descents).  Then it makes sense to set 'linespace' to a negative value.  This may cause display problems though!

```suggestion
    linespace: i64,
```